exporter/metric/internal/statsd/conn_test.go

```diff
@@ -0,0 +1,280 @@
+	return cagg
+}
+
+func measureAgg(desc *export.Descriptor, v float64) export.Aggregator {
+	ctx := context.Background()
+	magg := array.New()
+	_ = magg.Update(ctx, testNumber(desc, v), desc)
+	magg.Checkpoint(ctx, desc)
+	return magg
+}
+
+func TestBasicFormat(t *testing.T) {
+	for _, nkind := range []core.NumberKind{
+		core.Float64NumberKind,
+		core.Int64NumberKind,
+	} {
+		t.Run(nkind.String(), func(t *testing.T) {
+			ctx := context.Background()
+			writer := &testWriter{}
+			config := statsd.Config{
+				Writer:        writer,
+				MaxPacketSize: 1024,
+			}
+			adapter := newAdapter()
+			exp, err := statsd.NewExporter(config, adapter)
+			if err != nil {
+				t.Fatal("New error: ", err)
+			}
+
+			checkpointSet := test.NewCheckpointSet(adapter.LabelEncoder)
+			cdesc := export.NewDescriptor(
+				"counter", export.CounterKind, nil, "", "", nkind, false)
+			gdesc := export.NewDescriptor(
+				"gauge", export.GaugeKind, nil, "", "", nkind, false)
+			mdesc := export.NewDescriptor(
+				"measure", export.MeasureKind, nil, "", "", nkind, false)
+			tdesc := export.NewDescriptor(
+				"timer", export.MeasureKind, nil, "", unit.Milliseconds, nkind, false)
+
+			labels := []core.KeyValue{
+				key.New("A").String("B"),
+				key.New("C").String("D"),
+			}
+			const value = 123.456
+
+			checkpointSet.Add(cdesc, counterAgg(cdesc, value), labels...)
+			checkpointSet.Add(gdesc, gaugeAgg(gdesc, value), labels...)
+			checkpointSet.Add(mdesc, measureAgg(mdesc, value), labels...)
+			checkpointSet.Add(tdesc, measureAgg(tdesc, value), labels...)
+
+			err = exp.Export(ctx, checkpointSet)
+			require.Nil(t, err)
+
+			var vfmt string
+			if nkind == core.Int64NumberKind {
+				fv := float64(value)
+				vfmt = strconv.FormatInt(int64(fv), 10)
+			} else {
+				vfmt = strconv.FormatFloat(value, 'g', -1, 64)
+			}
+
+			require.Equal(t, 1, len(writer.vec))
+			require.Equal(t, fmt.Sprintf(`counter:%s|c|#A:B,C:D
+gauge:%s|g|#A:B,C:D
+measure:%s|h|#A:B,C:D
+timer:%s|ms|#A:B,C:D
+`, vfmt, vfmt, vfmt, vfmt), writer.vec[0])
+		})
+	}
+}
+
+func makeLabels(offset, nkeys int) []core.KeyValue {
+	r := make([]core.KeyValue, nkeys)
+	for i := range r {
+		r[i] = key.New(fmt.Sprint("k", offset+i)).String(fmt.Sprint("v", offset+i))
+	}
+	return r
+}
+
+type splitTestCase struct {
+	name  string
+	setup func(add func(int))
+	check func(expected, got []string, t *testing.T)
+}
+
+var splitTestCases = []splitTestCase{
+	// These test use the number of keys to control where packets
+	// are split.
+	{"Simple",
+		func(add func(int)) {
+			add(1)
+			add(1000)
+			add(1)
+		},
+		func(expected, got []string, t *testing.T) {
+			require.EqualValues(t, expected, got)
+		},
+	},
+	{"LastBig",
+		func(add func(int)) {
+			add(1)
+			add(1)
+			add(1000)
+		},
+		func(expected, got []string, t *testing.T) {
+			require.Equal(t, 2, len(got))
+			require.EqualValues(t, []string{
+				expected[0] + expected[1],
+				expected[2],
+			}, got)
+		},
+	},
```

Comment:
please add FirstBig (reverse of LastBig).

Reply:
Done. (There was a similar LastSmall case, but I added a new variation.)